synapse/handlers/typing.py

```diff
@@ -491,7 +491,7 @@ async def get_new_events_as(
                 ):
                     continue
 
-                events.append(self._make_event_for(room_id))
+                events.add(self._make_event_for(room_id))
```

Comment:
This feels like it being a set might not de-duplicate a lot since the object embeds a list of user IDs?